pkg/install/install.go

```diff
@@ -0,0 +1,467 @@
+package install
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/projectcalico/libcalico-go/lib/names"
+	"github.com/prometheus/common/log"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"k8s.io/client-go/rest"
+)
+
+type config struct {
+	// Location on the host where CNI network configs are stored.
+	CNINetDir   string `envconfig:"CNI_NET_DIR" default:"/etc/cni/net.d"`
+	CNIConfName string `envconfig:"CNI_CONF_NAME"`
+
+	// Directory where we expect that TLS assets will be mounted into the calico/cni container.
+	TLSAssetsDir string `envconfig:"TLS_ASSETS_DIR"`
+
+	// SkipCNIBinaries is a comma-separated list of binaries. Each binary in the list
+	// will be skipped when installing to the host.
+	SkipCNIBinaries string `envconfig:"SKIP_CNI_BINARIES"`
+
+	// UpdateCNIBinaries controls whether or not to overwrite any binaries with the same name
+	// on the host.
+	UpdateCNIBinaries bool
```

Comment:
nit:  Perhaps OverwriteHostBinaries is more accurate.  Does this need to be env controlled do you think?

Reply:
Good catch.Yes it is currently controlled by environment variable in shell script. Will fix it. Regarding name, maintained the same name used now UPDATE_CNI_BINARIES